lib/Sema/CSDiagnostics.cpp

```diff
@@ -8760,6 +8760,10 @@ bool MissingExplicitExistentialCoercion::fixItRequiresParens() const {
 
 void MissingExplicitExistentialCoercion::fixIt(
     InFlightDiagnostic &diagnostic) const {
+
+  if (ErasedResultType->hasDependentMember())
```

Comment:
Is hasDependentMember correct though? If so I think it's clearer if I say hasDependentMember() here just because it's easier to understand this check's purpose.  Seeing `hasDependentMember` helps visualize potential swift code that would not get printed whereas  `hasTypeParameter` just makes me think of an arbitrary type value

Reply:
I think it's okay to check for `hasTypParameter()` only because we don't want to suggest fix-its with unresolved type parameters either (although probably in this situations we cannot have plain type parameters in the type), it's just a broader check and implies `hasDependentMember` too.